ttnn/cpp/ttnn/operations/experimental/ccl/all_reduce/device/all_reduce_op.cpp

```diff
@@ -110,7 +121,7 @@ static Tensor all_gather_local_reduce(
     const std::optional<size_t> user_defined_num_workers,
     const std::optional<size_t> user_defined_num_buffers_per_channel,
     const std::vector<Device*>& devices,
-    const ttnn::ccl::Topology& topology) {
```

Comment:
Would it be better to keep the const guarantee and instead make a variable copy in the function? I am just worried that upstream this causes issues if the topology variable they want to feed in is already declared constant.

Reply:
Although this wasn't required for the fix, I noticed it was passed as const ref but it's just a simple enum so I changed it to a copy - so it is fine if the caller passes a const copy of this type.